### PR TITLE
Add a document describing the installation workflow

### PR DESCRIPTION
**Pull request by imobach**
This is a draft of a document describing (without going in-depth) the installation process. Sure it has a lot of obvious things but maybe it could be useful for newcomers.

### CODE REVIEW

File: doc/installation_overview.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+course, `autoyast` options.
+
+After the system is started, a script called `/sbin/inst_setup` is ran. That
+script performs some initialization tasks and, at the end, calls _YaST first
+stage_ (known as `initial`). This is the main part of the installation
+workflow: disks are set up (partitions and filesystem), software is installed,
+boot loader is prepared, etc. When the _first stage_ is over, the machine is
+rebooted to get into _YaST second stage_ (known as `continue`) if needed.
+
+That second stage is not always needed. As of SLE 12 and openSUSE 13.2, a regular
```
Comment (jreidinger) [inline doc/installation_overview.md]: maybe add "and later so it is obvious that also newer versions use this way

Comment (imobach) [inline doc/installation_overview.md]: :+1:

File: doc/installation_overview.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+repository](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Mode.rb)
+to learn more about that. This document is mainly about `installation` and
+`autoinstallation` modes.
+
+## Overview
+
+Although _YaST_ sits at the central point of the installation process, it is
+not the only software involved. Linuxrc, which will boot the system, is another
+key player.
+
```
Comment (aschnell) [inline doc/installation_overview.md]: AFAIK before linuxrc starts isolinux/syslinux is used to load the linux kernel. So I would say isolinux/syslinux boots the system.

Comment (imobach) [inline doc/installation_overview.md]: Ok. I'll make that point clear.

File: doc/installation_overview.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+to kick off the installation process (through `yast-installation`). After some
+initialization work (described below), this script goes through a series of
+steps defined scripts at `/usr/lib/YaST2/startup/First-Stage`.
+
+So the workflow is something like:
+
+* Import installation settings from `/etc/install.inf` as environment variables
+  (`import_install_inf` at `/usr/lib/YaST2/startup/common/misc.sh`).
+* Set up architeture variables (ARCH_ISERIES will be 1 on iseries architectures).
+* Start logging at `/var/log/YaST2/y2start.log`.
```
Comment (jreidinger) [inline doc/installation_overview.md]: I think it is not `start logging`, but actually `logging into`

File: doc/installation_overview.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+* Execute `preFirstStage` [hooks](https://github.com/yast/yast-yast2/blob/master/library/general/doc/Hooks.md).
+* Execute the `FirstStage` (see stages/steps). Additional `preFirstCall` and `postFirstCall` hooks are called.
+  _YaST_ gets called at this point.
+* Execute `postFirstStage` hooks.
```
Comment (jreidinger) [inline doc/installation_overview.md]: it also for upgrade restore original repository setup if offline update is canceled

File: doc/installation_overview.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+1. Hardware detection through `hwinfo`.
+2. Umount `inst-sys` (`/var/adm/mount`).
+3. Set up the language. Remember that you could specify a language through
+   the Linuxrc parameter `language`.
+4. Handle kernel parameters. A this time, it only copy `repair` to `install.inf` is present.
+5. Configure the terminal (and save TERM to `/etc/install.inf`).
+6. Set environment variables related to YaST log (size and max logs count).
+7. Start YaST in installation mode (`yast-installation`) with
+   `/usr/lib/YaST2/startup/YaST2.call installation initial`.
```
Comment (jreidinger) [inline doc/installation_overview.md]: do we always start as instalaltion? we have also update

File: doc/installation_overview.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+* Bringing up network (if configuration is found), starting a second shell and
+  getting prepared for a VNC installation (if needed).
+* Starting _YaST_ just if installation is going through _VNC_. If it must be done through
+  _SSH_, the user could start _YaST_ at this point.
+* Cleaning up things when installation is finished.
+
+#### inst_worker_continue
+
+The client which drives the second stage is `inst_worker_continue`. Of course, it relies on
+`ProductControl` as `inst_worker_initial` does.
```
Comment (jreidinger) [inline doc/installation_overview.md]: I prefer more links in this section to easier orientation when I need to inspect something in deep

**Comment by jreidinger** [comment]
otherwise it looks good for me.

File: doc/installation_overview.md (hunk 5 above)
Comment (imobach) [inline doc/installation_overview.md]: I think yes. AFAIK, `installation` is the client and the mode [is set up by `Mode` module](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Mode.rb#L108).

File: doc/installation_overview.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+`/usr/lib/YaST2/startup/YaST2.First-Stage`. The main purpose of this script is
+to kick off the installation process (through `yast-installation`). After some
+initialization work (described below), this script goes through a series of
+steps defined scripts at `/usr/lib/YaST2/startup/First-Stage`.
+
+So the workflow is something like:
+
+* Import installation settings from `/etc/install.inf` as environment variables
+  (`import_install_inf` at `/usr/lib/YaST2/startup/common/misc.sh`).
+* Set up architeture variables (ARCH_ISERIES will be 1 on iseries architectures).
```
Comment (jreidinger) [inline doc/installation_overview.md]: I am not sure if we need to mention it as we no longer support ISeries for ppc at least if I know.

File: doc/installation_overview.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+* Setting up logging (including `syslogd`).
+* If `/var/lib/autoinstall/autoconf/autoconf.xml` is found, then _AutoYaST_ is started
+  (as `/usr/lib/YaST2/startup/YaST2.call installation continue`)
+* Setting up language stuff (`LC_ALL` and `LANG` environment variables).
+* Setting up virtual console.
+* Starting services as `pcmcia` or `hald` (Hardware Abstraction Layer).
+* Bringing up network (if configuration is found), starting a second shell and
+  getting prepared for a VNC installation (if needed).
+* Starting _YaST_ just if installation is going through _VNC_. If it must be done through
+  _SSH_, the user could start _YaST_ at this point.
```
Comment (jreidinger) [inline doc/installation_overview.md]: what happen in non VNC and non SSH case? it start yast or not?

File: doc/installation_overview.md (hunk 7 above)
Comment (imobach) [inline doc/installation_overview.md]: Also maybe mentioning this is going into too much detail. Removed.

File: doc/installation_overview.md (hunk 8 above)
Comment (imobach) [inline doc/installation_overview.md]: I've misunderstood the `S03-autoconf` script. So yes, it's always started at this point.

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**